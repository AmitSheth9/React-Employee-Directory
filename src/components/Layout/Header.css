.container {
    display: flex;
    justify-content: space-evenly;
    border: 5px solid black;
}
.acme {
    background: yellow;
    color: brown;
    border: 2px solid black;
    font-size: 150%;
    width: 20%;
    
}