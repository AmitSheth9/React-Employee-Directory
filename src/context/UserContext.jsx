import { createContext, useContext, useState, useMemo } from "react";
import { getUser } from "../services/users";

const UserContext = createContext();

const UserProvider = ({ children }) => {
    const presentUser = getUser();
    const [user, setUser] = useState(presentUser ? { id: presentUser.id, email: presentUser.email } : {} );

    const [signedIn, setSignedIn] = useState(false);

    return <UserContext.Provider value={{ user, setUser, signedIn, setSignedIn }}>{children}</UserContext.Provider>
};

const useUser = () => {
    const context = useContext(UserContext);

    if (context === undefined) {
        throw new Error('useUser must be used inside UserProvider');
    }
    return context;
};

export { UserContext, UserProvider, useUser };