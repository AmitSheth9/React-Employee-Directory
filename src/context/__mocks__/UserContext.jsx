import { createContext, useContext, useState } from "react";


const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
    
    const [user, setUser] = useState(mockUser ? { id: mockUser.id, email: mockUser.email } : {} );

    const [signedIn, setSignedIn] = useState(false);

    return <UserContext.Provider value={{ user, setUser, signedIn, setSignedIn }}>{children}</UserContext.Provider>
};

const useUser = () => {
    const context = useContext(UserContext);

    if (context === undefined) {
        throw new Error('useUser must be used inside UserProvider');
    }
    return context;
};

export { UserContext, UserProvider, useUser };