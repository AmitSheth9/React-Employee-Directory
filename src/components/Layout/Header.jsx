import React from 'react'
import { Link } from 'react-router-dom';
import { useUser } from '../../context/UserContext'
import { signOutUser } from '../../services/users';
import { useHistory } from 'react-router-dom';
import styles from './Header.css'

export default function Header() {
    const auth = useUser();
    const history = useHistory();

    

    const handleClick = async () => {
        const userOut = await signOutUser();
        auth.setUser({});
        history.replace('/');
    }
    return (
        <div className={styles.container}>
            <div className={styles.acme}>
                <Link to='/'>ACME INC</Link>
            </div>
            <div className={styles.status}>
            {auth.user.email ? `Welcome, ${auth.user.email}` : ''}
            </div>
            <div>
            {auth.user.email ? <button onClick={handleClick}>Sign Out</button> : <div className={styles.authstatus}>  <Link to='/login'>  Sign in</Link></div> }
            
            <br/><br/>
            </div>
            <br/><br/>
        </div>
    )
}
