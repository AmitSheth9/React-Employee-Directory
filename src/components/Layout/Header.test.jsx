import {render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import Header from './Header';

jest.mock('../../services/users')
jest.mock('../../context/UserContext')

it('should render Header', () => {
    const { container } = render(
        <UserProvider>
       
        <MemoryRouter>
            <Header />
        </MemoryRouter>
       
        </UserProvider>
    );
    expect(container).toMatchSnapshot();
} );