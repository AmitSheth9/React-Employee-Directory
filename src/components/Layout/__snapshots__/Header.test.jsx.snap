// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="acme"
    >
      <a
        href="/"
      >
        ACME INC
      </a>
    </div>
    <div>
      <div>
        Not Signed In  
        <a
          href="/login"
        >
            Sign in
        </a>
      </div>
    </div>
    <div>
      <p />
    </div>
    <br />
    <br />
  </div>
</div>
`;
